### PR TITLE
add a paragraph about environment variables

### PR DESCRIPTION
**Pull request by cake17**
Refs to https://github.com/cakephp/app/pull/328, https://github.com/cakephp/app/issues/224 and https://github.com/cakephp/docs/issues/3674

Let me know if this is the right place in the docs ? And don't hesitate to correct my english :smile:

### CODE REVIEW

File: en/development/configuration.rst [hunk 1]
```diff
@@ -614,6 +614,41 @@ order to use them in your controllers. As you'll see in the :doc:`/controllers`
...
+- APP_DEFAULT_LOCALE (ie en_US)
+- SECURITY_SALT
+- CACHE_DEFAULT_URL (ie File:///?prefix=myapp_&serialize=true&timeout=3600&path=../tmp/cache/)
+- CACHE_CAKECORE_URL (ie File:///?prefix=myapp_cake_core_&serialize=true&timeout=3600&path=../tmp/cache/persistent/)
+- CACHE_CAKEMODEL_URL (ie File:///?prefix=myapp_cake_model_&serialize=true&timeout=3600&path=../tmp/cache/models/)
+- EMAIL_TRANSPORT_DEFAULT_URL (ie smtp://user:password@hostname:port?tls=null&client=null&timeout=30)
+- DATABASE_URL (ie mysql://user:pass@db/my_app)
+- DATABASE_TEST_URL (ie mysql://user:pass@db/test_my_app)
+- LOG_DEBUG_URL (ie file:///?levels[]=notice&levels[]=info&levels[]=debug&file=debug&path=../logs/)
+- LOG_ERROR_URL (ie file:///?levels[]=warning&levels[]=error&levels[]=critical&levels[]=alert&levels[]=emergency&file=error&path=../logs/)
```
Comment (markstory) [inline en/development/configuration.rst]: Should these config keys be inline-code blocks?

**Comment by josegonzalez** [comment]
Yay you wrote the thing I forgot to write!

File: en/development/configuration.rst (hunk 1 above)
Comment (josegonzalez) [inline en/development/configuration.rst]: And the sample values...

Comment (cake17) [inline en/development/configuration.rst]: @markstory You mean I should add indentation ?
```
    - DEBUG=0
    - APP_ENCODING=UTF-8
    - APP_DEFAULT_LOCALE=en_US
    - SECURITY_SALT=__SALT__
    - CACHE_DEFAULT_URL=null
    - CACHE_CAKECORE_URL=null
    - CACHE_CAKEMODEL_URL=null
    - EMAIL_TRANSPORT_DEFAULT_URL=null
    - DATABASE_URL=null
    - DATABASE_TEST_URL=null
    - LOG_DEBUG_URL=null
    - LOG_ERROR_URL=null
```
@josegonzalez you'd prefer the default sample values like above ?

Comment (markstory) [inline en/development/configuration.rst]: @cake17 More that the generated output would look like:

- `CACHE_DEFAULT_URL` (ie `file:///?prefix=myapp_&serialize=true&timeout=3600&path=../tmp/cache/`)

**Comment by cake17** [comment]
@markstory Inline-code blocks added. I also added the french part

**Pull request merged**